### PR TITLE
fix: mm mobile browser pairing

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Fixes borked MM mobile browser pairing.
While at it
- fixes inconsistent deep-linking behavior, which sometimes work, sometimes does not depending on the browser and other non-deterministic things like current weather conditions
- doesn't display the MM snaps modal on MM mobile since snaps aren't available on mobile

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/5884

## Risk

Not a risk per se, but deep-linking isn't a deep link anymore, but rather a proper app URL, see https://github.com/MetaMask/metamask-mobile/pull/4167. While this makes "deep-linking" *consistently* working vs. opening the app store even though the app is installed, the downside is users now won't be linked to the app store if the MM app isn't installed on their device.

## Testing

- When opening the app from MM browser, it is possible to pair MM
- When opening the app from a mobile browser (tested with Orion, for which the deep link didn't work), it is possible to open the app in MM by tapping "Open"

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

### Prod

IMG_4262

### This diff
 
IMG_4263

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #5887** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5887?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5887?utm_source=stack-comment).

File: src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx [hunk 1]
```diff
@@ -35,6 +35,16 @@ export const checkIsMetaMask = pMemoize(
...
+export const checkisMetaMaskMobileWebView = () => {
+  if (typeof window === 'undefined') {
+    return false
+  }
+
+  return (
+    Boolean(window.ReactNativeWebView) && Boolean(navigator.userAgent.endsWith('MetaMaskMobile'))
+  )
+}
+
```
Comment (gomesalexandre) [inline src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx:47]: Copied from

https://github.com/MetaMask/metamask-sdk/blob/6230d8394157f53f1b020ae44601a0a69edc6155/packages/sdk/src/Platform/PlatfformManager.ts#L102C30-L111

File: src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx [hunk 2]
```diff
@@ -35,6 +35,16 @@ export const checkIsMetaMask = pMemoize(
   },
 )
 
+export const checkisMetaMaskMobileWebView = () => {
```
Comment (gomesalexandre) [inline src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx:38]: Now that we have this, we are able to disable the snaps modal showing up on MM mobile

File: src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx
Comment (gomesalexandre) [inline src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx:1]: Note, this house looks weird, but probably the best place for it alongside other MM detection utils

File: src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx:39]: 91439d2cc11e3be4905aa39b500990025c0f8754

**Review by woodenfurniture — COMMENTED** [review]
Tested on `gome`, working as intended with 1 exception (detailed below)

Pairing from mobile brave browser connects to metamask as intended:
Screenshot_20231219-131636
Screenshot_20231219-131646

Pairing from metamask browser working, but the "pair" modal keeps re-appearing and you cant actually use the app unless you close the modal. Metamask is connected after closing the pair modal:
Screenshot_20231219-131918
Screenshot_20231219-131935
Screenshot_20231219-131943
Screenshot_20231219-132009

**Review by woodenfurniture — APPROVED** [review]
confirmed latest changes solved the minor issue with pairing modal, lgtm!

**Pull request merged**